vulcanai2/plotters/visualization.py

```diff
@@ -21,17 +20,56 @@
 import logging
 logger = logging.getLogger(__name__)
 
+def assert_display_available():
+    """
+    Check that display variable exists
+
+    Returns
+    -------
+    Boolean
+        True if display variable exists
+        False if display variable does not exist
+
+    """
+    return True if os.environ.get("DISPLAY") else False
 
-def display_record(record=None, load_path=None, interactive=True):
+def save_visualization(plot, path=None):
+    """
+    Save plot at designated path
+
+    Parameters
+    ----------
+    plot : matplotlib
+        Matplotlib variable with savefig ability.
+    path : string
+        String that designates the path to save the given figure to.
+
+    Returns
+    -------
+    None
+
+    """
+    if not assert_display_available() and path is None:
```

Comment:
Wait, but what if the path is None but the display is available?

Reply:
In general this doesn't seem relevant to a "save_visualization" function. Why is this here?

Reply:
@nirtiac @rfratila I know we had a reason, but I can't seem to remember it. However, looking at it now it doesn't make sense. If a path is specified we need to save the visualization before plotting and we need to confirm a display variable exists before plotting. We can remove that `assert_display_available()` from the `save_visualization` function and put it in an if statement before `plt.show()` for each display function.